tests/python/test_texture.py

```diff
@@ -103,8 +103,10 @@ def paint(t: ti.f32, tex: ti.types.texture(num_dimensions=2), n: ti.i32):
     paint(0.2, texture2, n2)
     assert impl.get_runtime().get_num_compiled_functions() == 3
 
+    # (penguinliong) Remember that non-RW textures don't enforce a format so
+    # it's the same as the first call to `paint`.
     paint(0.3, texture3, n1)
-    assert impl.get_runtime().get_num_compiled_functions() == 4
+    assert impl.get_runtime().get_num_compiled_functions() == 3
```

Comment:
nit: shall we also assert on sampled `pixels` values to make sure it's numerically correct? IIRC we don't have such test yet.

Reply:
shouldn't this be covered by the texture aot example?